pdf/pdflinkarea.cpp

```diff
@@ -148,8 +192,50 @@ void PDFLinkArea::mouseReleaseEvent(QMouseEvent *event)
     if (!rect.contains(event->pos()))
         return;
 
+    // Click action logic in order.
+    // - click on selection;
+    // - unselect if selection is set;
+    // - click on annotation;
+    // - click on link;
+    // - click.
+
     if (!d->canvas) {
         emit clicked(d->clickLocation);
+    } else if (d->clickOnSelection) {
+        emit selectionClicked();
+    } else if (d->selection && d->selection->count() > 0) {
+        d->selection->unselect();
+    } else if (d->annotation != nullptr) {
+        // proxy will be child of this object to avoid memory leak.
```

Comment:
That's right. At the time I wrote this, I read extensively the Qt documentation and I remember reading that transfer ownership to QML was impossible: http://doc.qt.io/qt-5/qtqml-cppintegration-data.html <quote>When data is transferred from C++ to QML, the ownership of the data always remains with C++.</quote>

So yes, all the annnotation wrappers, generated when tapping on an annotation, will stay alive for the duration of the document. I don't think that's really a memory penalty, even on small memory systems. Indeed, the wrapper is only sizeof(QObject) + some pointers and one int and one boolean. So we're speaking of some kilobytes at most even if the user is tapping on dozen of annotations. Don't you think so ?